File: Ontology/ObjectPropertyAxiom.cs
﻿namespace Ontology
{
    public abstract class ObjectPropertyAxiom:
        Axiom,
        IObjectPropertyAxiom
    {
        private readonly IObjectPropertyExpression _objectPropertyExpression;

        protected ObjectPropertyAxiom(
            IOntology                 ontology,
            IObjectPropertyExpression objectPropertyExpression
            ) : base(ontology)
        {
            _objectPropertyExpression = objectPropertyExpression;
        }

        IObjectPropertyExpression IObjectPropertyAxiom.ObjectPropertyExpression => _objectPropertyExpression;
    }
}

File: Web/ClientApp/src/app/Ontology/ObjectPropertyAxiom.ts
import { Axiom } from "./Axiom";
import { IObjectPropertyAxiom } from "./IObjectPropertyAxiom";
import { IOntology } from "./IOntology";

export class ObjectPropertyAxiom
    extends Axiom
    implements IObjectPropertyAxiom
{
    constructor(
        ontology: IOntology
        )
    {
        super(ontology);
    }
}
